Add yolink integration document (#22688)
home-assistant/home-assistant.github.io: source/_integrations/yolink.markdown

---
title: YoLink
description: Instructions on how to integrate YoLink Devices into Home Assistant.
ha_category:
  - Binary Sensor
  - Sensor
  - Siren
  - Switch
ha_release: 2022.6
ha_iot_class: Cloud Push
ha_config_flow: true
ha_codeowners:
  - '@YoSmart-Inc'
ha_domain: yolink
ha_platforms:
  - binary_sensor
  - sensor
  - siren
  - switch
ha_integration_type: integration
---

Integrates [YoLink](https://www.yosmart.com/) Devices into Home Assistant.

{% include integrations/config_flow.md %}

> Unless configured otherwise, Home Assistant will use account linking provided by
> Nabu Casa for authenticating with YoLink, If you want to use separate credentials, please contact <service@yosmart.com> to obtain a `client_id` and `client_secret`. Then you can add your credentials via application credentials. Settings > Devices & Services > click the menu (three dots at the top right of the screen) and then **applicaton Credentials**. Enter your credentials in the pop-up window.

Reviewer: ```suggestion
Unless configured otherwise, Home Assistant will use account linking provided by
Nabu Casa for authenticating with YoLink, If you want to use separate credentials, please contact <service@yosmart.com> to obtain a `client_id` and `client_secret`. Then you can add your credentials via application credentials. Settings > Devices & Services > click the menu (three dots at the top right of the screen) and then **Application Credentials**. Enter your credentials in the pop-up window.
```

Author: done

---
title: YoLink
description: Instructions on how to integrate YoLink Devices into Home Assistant.
ha_category:

>   - yolink

Reviewer 2: Please only use the existing categories. See other integration pages for examples.

> ha_release: 2022.5.4

Reviewer 2: ```suggestion
ha_release: 2022.6.0
```

> ha_release: 2022.6.0

Reviewer: ```suggestion
ha_release: 2022.6
```

ha_iot_class: Cloud Push
ha_config_flow: true
ha_codeowners:
  - '@YoSmart-Inc'
ha_domain: yolink
ha_platforms:

>   - yolink

Reviewer 2: ```suggestion
  - sensor
```

ha_integration_type: integration
---

Integrates [YoLink](https://www.yosmart.com/) Devices into Home Assistant.

> You must contact <service@yosmart.com> to obtain a `client_id` and `client_secret`.

Reviewer: Is this still true? We use account linking now, right?

> You must concat us <service@yosmart.com> to obtain a `client_id` and `client_secret`.

Reviewer 2: ```suggestion
You must contact <service@yosmart.com> to obtain a `client_id` and `client_secret`.
```

To enable this platform, add the following lines to your `configuration.yaml` file:

> 

Reviewer 2: We should update the docs now and mention the application credentials in the frontend and Home Assistant Cloud Account Linking.

```yaml
# Example configuration.yaml entry
yolink:
  client_id: CLIENT_ID
  client_secret: CLIENT_SECRET
```

Restart Home Assistant. Then, go to the frontend and authorize YoLink.

{% include integrations/config_flow.md %}

{% configuration %}
client_id:
  description: Your YoLink API client ID.
  required: true
  type: string
client_secret:
  description: Your YoLink API client secret.
  required: true
  type: string

> {% endconfiguration %}

Reviewer: This has been replaced by application credentials.

Author: this is no longer necessary, I will remove that, thanks

* YS7704

> {% include integrations/config_flow.md %}

Reviewer 2: I'd move this include above the configuration specification as the include contains a header for configuration.

Integrates [YoLink](https://www.yosmart.com/) Devices into Home Assistant.

{% include integrations/config_flow.md %}

Unless configured otherwise, Home Assistant will use account linking provided by

> Nabu Casa for authenticating with YoLink, If you want to use separate credentials, Please contact <service@yosmart.com> to obtain a `client_id` and `client_secret`, Then you can add your credentials via application credentials.

Reviewer 2: ```suggestion
Nabu Casa for authenticating with YoLink. If you want to use separate credentials, please contact <service@yosmart.com> to obtain a `client_id` and `client_secret`. Then you can add your credentials via application credentials.
```

Reviewer 2: Maybe we need to say where to find the application credentials setting? @frenck what do you think?
